main.js

```diff
@@ -1,47 +1,38 @@
+'use strict';
+
 var React;
 
-if (window) {
+if (typeof window !== 'undefined') {
   // circumvent React Dev Tools console warning
   window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = {};
 }
 
-function _injectCustomConfig() {
-  var ReactInjection = require('react/lib/ReactInjection');
-  var customAttributes = {
-    mini: ReactInjection.DOMProperty.HAS_BOOLEAN_VALUE
-  };
-  var AtomDOMPropertyConfig = {
-    isCustomAttribute: function(attributeName) {
-      return customAttributes[attributeName] !== undefined;
-    },
-    Properties: customAttributes
-  };
-  ReactInjection.DOMProperty.injectDOMPropertyConfig(AtomDOMPropertyConfig);
-}
-
 if (typeof atom === 'object' && atom !== null) {
-  if (atom.__DO_NOT_ACCESS_React_Singleton === undefined) {
-    React = require('react');
-    _injectCustomConfig();
-    var addons;
-    Object.defineProperty(React, 'addons', {
-      get: function get() {
-        if (!addons) {
-          React = require('react/addons');
-        }
-        return addons;
-      },
-      set: function set(val) {
-        addons = val;
-      },
-      enumerable: false,
-    });
+  if (typeof atom.__DO_NOT_ACCESS_React_Singleton === 'undefined') {
+    if (process.env.NODE_ENV === 'production') {
+      React = require('react/dist/react-with-addons.min');
+    } else {
+      React = require('react');
+      var addons;
+      Object.defineProperty(React, 'addons', {
+        get: function get() {
+          if (!addons) {
+            React = require('react/addons');
```

Comment:
Correct. I added here for parity with React, not really caring much because we're not going to be using the addons much longer. I didn't flat out require this version to begin with because you'd see a warning, even if you don't use the addons.

@jgebhardt: @ssorallen and I talked IRL and we think it makes more sense to just require the addons straight out instead. wdyt?